### ISSUE TITLE
Dependency management

### ISSUE THREAD

**Issue by lk16** [skepticoin/skepticoin#69]
I noticed for a while the requirements file has no exact versions, This may lead to unexpected results when a new (major) version of a dependency is released. Also difference of dependencies on two machines may make bugs harder to reproduce.

On a somewhat related note, what do you guys think about switching from pip to [poetry](https://python-poetry.org/):
* Supports showing a dependency tree.
* Supports development dependencies 
* Ease of publishing to pypi: `poetry publish --build`
* ...

**Comment by atramos** [comment]
I agree we should add specific versions to the requirements.txt file.

I don't have an opinion on whether poetry is worth the extra weight over pip.

**Comment by lk16** [comment]
While release 0.1.17 (see #98), I noticed that poetry allows to just `poetry publish` instead of going through [this 5 step plan](https://stackoverflow.com/a/52727549) with a few pitfalls.

**Comment by sashimi-houdini** [comment]
I'm happy to pin the requirements to some specific version and I would take a PR that does so immediately. Switching to poetry isn't worth it for me
